lib/importer/repository.rb

```diff
@@ -0,0 +1,34 @@
+module Importer
+  class Repository
+    def self.process(text, source)
+      repos = text.scan(/(?:https?\:\/\/)(#{source})\/
+        ([^),.\/]+)\/([^,\s)(\/]+)\/?([^ )]*)/x)
+
+      repos.map do |repo|
+        Repository.new(repo).process
+      end
+    end
+
+    def initialize(repo)
+      @source = repo[0]
+      @username = repo[1].gsub(/\p{Z}/, '')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.